gnomad/utils/annotations.py

```diff
@@ -1396,75 +1404,104 @@ def _create_group_dicts(
             "popMaxFAF95": {
                 "frequency": ht.popmax.faf95.collect()[0],
                 "confidenceInterval": 0.95,
-                "popFreqID": f"{variant}.{ht.popmax.pop.collect()[0].upper()}",
+                "popFreqId": f"{variant}.{ht.popmax.pop.collect()[0].upper()}",
             },
             "homozygotes": overall_freq["homozygote_count"].collect()[0],
-            "meanDepth": coverage_stat,
+            "meanDepth": mean_coverage,
         },
     }
 
-    if groups:
-        ret_final["subpopulationFrequency"] = list_of_group_info_dicts
+    # If ancestry_groups were passed, and the dictionary which would have ben created to the dictionary to be returned
+    if ancestry_groups:
+        final_freq_dict["subpopulationFrequency"] = []
+        for info_dict in list_of_group_info_dicts:
+            final_freq_dict["subpopulationFrequency"].append(info_dict)
 
+    # Validate that our constructed dict would validate to a valid JSON
     try:
-        validated_json = json.dumps(ret_final)
+        validated_json = json.dumps(final_freq_dict)
     except:
         raise SyntaxError("The dictionary did not convert to a valid JSON")
 
-    return ret_final
+    # Returns the constructed dictionary.
+    return final_freq_dict
 
 
 def gnomad_gks(
     version: str,
     variant: str,
-    groups: list = None,
+    data_type: str = "genomes",
+    coverage_version: str = "3.0.1",
```

Comment:
rather than passing in a coverage version, grab high_level_version (see edit below) and load the coverage table based on that